yt/loaders.py

```diff
@@ -369,9 +369,9 @@ def load_uniform_grid(
                 bbox,
                 cell_widths=cell_widths,
             )
-            cell_widths = grid_cell_widths
             grid_dimensions = np.array(list(shapes), dtype="int32")
             temp[key] = [data[key][slice] for slice in slices]
+        cell_widths = grid_cell_widths
```

Comment:
I don't think I understand what the problem is or why this fixes it.

Reply:
oops, sorry, should have included more details. the fix was obvious to me, but only after like 45 mins of debugging lol. 

The following loop when `nprocs > 1`: (1) takes the continuous data arrays and splits them into separate grids and (2) records the grid edges, grid dimensions, and cell widths (if applicable) for each grid: 

```python
    if nprocs > 1:
        temp = {}
        new_data = {}  # type: ignore [var-annotated]
        for key in data.keys():
            psize = get_psize(np.array(data[key].shape), nprocs)
            (
                grid_left_edges,
                grid_right_edges,
                shapes,
                slices,
                grid_cell_widths,
            ) = decompose_array(
                data[key].shape,
                psize,
                bbox,
                cell_widths=cell_widths,
            )
            grid_dimensions = np.array(list(shapes), dtype="int32")
            temp[key] = [data[key][slice] for slice in slices]
            # setting cell_widths = grid_cell_widths would overwrite for next loop = bad
        cell_widths = grid_cell_widths
```

Only one copy of the grid related variables (grid_dimensions, grid_left_edges, grid_right_edges, cell_widths) is needed for the subsequent call to `load_amr_grids` below. When there are multiple fields to split, those grid related variables just get re-created every loop and only the last one is kept. The final `grid_cell_widths` needs to be renamed to `cell_widths` to match the behavior down in the rest of the function, but that rename has to happen outside the loop over fields or the call to `decompose_array` would use a cell_widths that was already split by grid.

Reply:
Oh, I see it now. I somehow missed that `cell_widths` would be fed back to `decompose_array`. Thanks a lot !